lib/CrossReferenceFolder.cpp

```diff
@@ -335,10 +335,11 @@ CrossReferenceFolderImpl::ResolveInstruction(llvm::Instruction *inst_val) {
     case llvm::Instruction::IntToPtr: {
       xr = ResolveValue(inst_val->getOperand(0));
       xr.size = static_cast<unsigned>(out_size);
-      if (auto ptr_type = llvm::cast<llvm::PointerType>(inst_val->getType());
-          !xr.displacement_from_hinted_value_type) {
-        xr.hinted_value_type = ptr_type->getElementType();
-      }
+      // NOTE(alex): Looks like this just improves the fidelity of the lift but not correctness?
```

Comment:
This is the case for a lot of blocks I've commented out. I'd be interested to know which ones are reasonable to remove and which ones need to remain (and have the underlying pointer type propagated somehow).

Reply:
I'd liek to see what happens if you delete `hinted_value_type` altogether 😈

Reply:
And as a consequence, anything reliant on it.

Reply:
> I'd like to see what happens if you delete `hinted_value_type` altogether 😈

I'll give that a try.

Reply:
[6e4821e](https://github.com/lifting-bits/anvill/pull/296/commits/6e4821e3587b60892bc5c3c0b999874d9a1c5151)